workers/gossip-server/src/peer_discovery/peers.rs

```diff
@@ -51,6 +52,50 @@ impl GossipProtocolExecutor {
         Ok(GossipResponse::Heartbeat(resp))
     }
 
+    /// Handle a proposed expiry from a peer
+    ///
+    /// TODO: Update this logic, for now we simply check if the local peer
+    /// thinks the expiry candidate should expire
+    pub async fn handle_propose_expiry(
+        &self,
+        sender: WrappedPeerId,
+        peer_id: WrappedPeerId,
+    ) -> Result<(), GossipError> {
+        let peer_info = self.global_state.get_peer_info(&peer_id).await?;
+        let info = match peer_info {
+            Some(info) => info,
+            None => return Ok(()),
+        };
+
+        // If the local peer has received a recent heartbeat from the candidate, notify
+        // the sender that the expiry should not proceed
+        let now = get_current_time_millis();
+        let time_since_last_heartbeat = now - info.last_heartbeat;
+        if time_since_last_heartbeat < HEARTBEAT_FAILURE_MS {
```

Comment:
not sure if we're planning on adjusting these windows later, but for now should this be check be using `CLUSTER_HEARTBEAT_FAILURE_MS` depending on if the peer is a cluster peer?

Reply:
yeah i just filled this in to test, the actual logic is more complex than either